samcli/lib/package/artifact_exporter.py

```diff
@@ -89,11 +89,13 @@ def do_export(self, resource_id, resource_dict, parent_dir):
             temporary_file.write(exported_template_str)
             temporary_file.flush()
 
-            url = self.uploader.upload_with_dedup(temporary_file.name, "template")
+            url = self.uploader.get(self.EXPORT_DESTINATION).upload_with_dedup(temporary_file.name, "template")
 
             # TemplateUrl property requires S3 URL to be in path-style format
             parts = parse_s3_url(url, version_property="Version")
-            s3_path_url = self.uploader.to_path_style_s3_url(parts["Key"], parts.get("Version", None))
+            s3_path_url = self.uploader.get(self.EXPORT_DESTINATION).to_path_style_s3_url(
```

Comment:
This needs to be reverted too.

Reply:
Sorry, I brought the confusion by https://github.com/aws/aws-sam-cli/pull/2440#issuecomment-737715257. The https://github.com/aws/aws-sam-cli/pull/2440/commits/0214f4b440de77785fe5723e0ec866525f6653dc seems all we have to change.

Reply:
Yep, on further testing, only have https://github.com/aws/aws-sam-cli/commit/0214f4b440de77785fe5723e0ec866525f6653dc in place now.